### DISCUSSION TITLE
Can I achieve alignment like `gofmt`?

### DISCUSSION THREAD

**Discussion by AuroraTea** [prettier/prettier#16274] [General]
Like this:
```JavaScript
const myObject = {
  short:          "value1",
  longerName:     "value2",
  evenLongerName: "value3",
};
```

**Comment by kachkaev** [comment]
There is no such option, see https://github.com/prettier/prettier/issues/350 and https://github.com/prettier/prettier/issues/1622 for context. You can add `// prettier-ignore` as a one-off trick:

**Prettier 3.2.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACADgJzjGASzhwFpCBzKCPAHSkigGcYMBbATwHkAjAKzhhWAXgzB6GDEwAWNGEkmKlGWiABuAQwA2AVzgBGVQBoJGLdAokAchrZwFS1Zt1wATMdNw1CADIXrtvYq6tp6AMyqSiZQAL4A3PT0jCzs3DDSJLwCQhii4lCSMnIOyopOoQYeBWb+ODZ2JZLlLu4g0ZJevrX1Qc3hkYrR8W0gEFhE0EzIoBo4OBAA7gAKswhTKNoLGhxTRiA8OBpgANb4AMqBPoRQcMgAZtpMcHsHR6cwZ1hH1xTIMDh6PZPNiEP4A54gOAADywJEIdlg2gAKiQoLNiOsHlonkCflo4ABFHQQeD3R4QvhMKFnPGE4mkpBYnEgACO9LgS3mWHWIA0TFINzgABNhSN-hpCFofgBhCBsNgaZC8rRaEZMWkAQQIOEIPB08CWJCuNzJ2Ih0hgbC0AHVpIR4EwvmA4Gc1vbCGp7RwlWAmLt1HoAJJQEWwM5gHXjDUhs4wDj403M3AQJ7Ww5YJW4OBPHDeEbXHMwTkaCgKxMQr44HNKngaHhwVV7XDXGDWwhC9LIAAcAAY9ng2YQ8MXS4rGeS9jA622O9JkK49jonki65iJ5C2PWhSKhT4NFAKDoS3AAGI0BUEH5KjT6iAgGIxIA)

```sh
--parser babel
```

**Input:**

```jsx
// prettier-ignore
const myObject = {
  short:          "value1",
  longerName:     "value2",
  evenLongerName: "value3"    ,
};

const myOtherObject = {
  short:          "value1",
  longerName:     "value2",
  evenLongerName: "value3"    ,
};
```

**Output:**

```jsx
// prettier-ignore
const myObject = {
  short:          "value1",
  longerName:     "value2",
  evenLongerName: "value3"    ,
};

const myOtherObject = {
  short: "value1",
  longerName: "value2",
  evenLongerName: "value3",
};

```

Note that `// prettier-ignore` turns off all formatting for the following statement, so it’s not just affecting value alignment.

Aligned values look neat in a ’static’ code snapshot, but they don’t play nice in an evolving codebase. Imagine we want to add `evenMuchLongerName: "value4"` to `myObject`. Because all three existing lines need to change, the diff becomes unreasonably big:

```diff
 const myObject = {
-  short:          "value1",
-  longerName:     "value2",
-  evenLongerName: "value3",
+  short:              "value1",
+  longerName:         "value2",
+  evenLongerName:     "value3",
+  evenMuchLongerName: "value4",
 };
 ```
 
 The above diff is harder to review compared to:
 
 ```diff
 const myOtherObject = {
   short: "value1",
   longerName: "value2",
   evenLongerName: "value3",
+ evenMuchLongerName: "value4",
 };
```
